api/src/main/java/io/opentelemetry/trace/DefaultSpan.java

```diff
@@ -77,7 +78,7 @@ public void addEvent(String name, Attributes attributes) {}
   public void addEvent(String name, Attributes attributes, long timestamp) {}
 
   @Override
-  public void setStatus(Status status) {}
+  public void setStatus(StatusCanonicalCode canonicalCode, @Nullable String description) {}
```

Comment:
I would use two methods instead of requiring null when no description, similar to how we handle exception / link attributes.

Reply:
I left a TODO, will do in the next PR https://github.com/open-telemetry/opentelemetry-java/pull/1741/files#diff-cc14ea515ee096dbd8ed1e630e45745fR217